pkg/kubectl/apply.go

```diff
@@ -188,3 +201,34 @@ func CreateOrUpdateAnnotation(createAnnotation bool, info *resource.Info, codec
 	}
 	return UpdateApplyAnnotation(info, codec)
 }
+
+// restoreJSONNulls is used to restore missing null values declared at the user's configuration file.
+// We do not expect conflicts.
+// TODO(andronat): Do we really need dataStruct? Is there any scenario where 'patchStrategy' might create problems?
+func restoreJSONNulls(j1, j2 []byte, dataStruct interface{}) ([]byte, error) {
```

Comment:
style: This probably belongs in strategicpatch (if it survives review and your TODOs)